Importer/Interpreter/EntityDataInterpreter.php

```diff
@@ -102,40 +105,36 @@ public function __construct(
         $this->validator = $validator;
         $this->entityManager = $entityManager;
         $this->eventDispatcher = $eventDispatcher;
+        $this->log = $log;
     }
 
     /**
      * @param array $data
+     * @param int   $maxErrorAmount
      * @param bool  $associative
      *
-     * @return object[]
-     * @throws InterpreterException
+     * @return array
      */
-    public function interpret(array $data, $associative = true)
+    public function interpret(array $data, $maxErrorAmount, $associative = true)
     {
         $entities = [];
         foreach ($data as $index => $row) {
+            if ($this->log->getErrorCount() >= $this->log->getMaxErrorCount()) {
+                break;
+            }
             try {
                 $entity = $this->interpretRow($row, $associative);
                 $entities[$index] = $entity;
                 $this->internalLookupCache[] = $entity;
             } catch (InterpreterException $exception) {
-                $this->handleInterpreterError($exception, $index, $row);
+                $this->eventDispatcher->dispatch('EVENT_DATA_ERROR', new DataErrorEvent($exception, $index, $this->log));
```

Comment:
It might be more readable to put this `if` condition here, just after dispatching an event.